src/java.base/share/classes/java/lang/foreign/MemorySegment.java

```diff
@@ -609,25 +602,24 @@ public sealed interface MemorySegment permits AbstractMemorySegmentImpl {
     long byteSize();
 
     /**
-     * {@return the <em>maximum</em> byte alignment (which is equal to or higher than
-     * the requested byte alignment during native segment allocation)}
+     * {@return the <a href="#segment-alignment">maximum byte alignment</a>
+     * associated with this memory segment}
      * <p>
-     * The returned alignment is always an even power of two and is derived from:
+     * The returned alignment is always a power of two and is derived from:
      * <ul>
      *     <li>Heap:
-     *     the segment offset and backing array type.</li>
+            derived from the segment {@linkplain #address() address()}
+            and the type of the <a href="#segment-alignment">backing heap storage</a>.
```

Comment:
Shoudn't the link point to `MemorySegment::heapObject` ?

Reply:
I suggest you generate the javadoc and look at this method - my feeling is that it won't be very readable. The sentence "a power of two and is derived from" is left a bit suspended - and you have a bullet list which starts with "Native Segment" which is not a valid "next token" for the sentence above. After which, we say again "derived from" hence repeating.